core/src/rpc.rs

```diff
@@ -1226,6 +1259,20 @@ fn verify_token_account_filter(
     }
 }
 
+fn check_slice_and_encoding(encoding: &UiAccountEncoding, data_slice_is_some: bool) -> Result<()> {
+    if data_slice_is_some && encoding == &UiAccountEncoding::JsonParsed {
```

Comment:
I'd prefer a `match` on each enum item of UiAccountEncoding, so we get a compile error here when a new encoding is added.